string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,49 @@
+/* exported titleCase */
+
+// make a storage for output (empty string)
+// make title to array form, splitting them for spaces in between words
+// look at each item of the title in array form, beginning until the end
+// if any of the items lowercased is 'api'
+// replace that item with 'API'
+// also, if any of the items lowercased is 'javascript'
+// replace that item with 'JavaScript'
+// also, if the length of an item is greater than 3
+// replace that item with the first letter capitalized, rest is lowercased
+// also, if the any of the item's next item is 'developers'
+// replace that item with the first letter capitalized, rest is lowercased
+// make a storage for modded title, and assign that value and convert it back to string form
+// look at each item of the modded title, beginning until the end
+// if any of the item is equal to ':'
+// replace the second item right of ':' and uppercase
+// also, if any of the item is equal to '-'
+// replace the item right of '-' and uppercase
+// uppercase the first letter of the modded title, and assign it to output storage
+// give back output
```

Comment:
I think this will be a lot easier if you make an array of the "minor words" listed in the rules. Then you'll be able to check if a word in the title is in that array.

Right now your solution has a lot of assuming that a 3-letter word should be lowercased, but lots of 3-letter words are not minor words.